### PR TITLE
Burning Building fix

### LINKED ISSUE #11319: Burning Building scenario fire fields incorrectly placed
In the Burning Building scenario, sometimes (but not always), the fire fields are outside of the actual building. Here's an example (start building is a Library):
2015-02-18 - 02_52_12pm

I've uploaded the save file where this happened:
http://www.mediafire.com/download/03y2kekhytc5616/Roaming_Shores.zip

Cata version: 14fea4514ebb5d161de81473ffd1aec5e2ab5aab

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/8691399-burning-building-scenario-fire-fields-incorrectly-placed?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F146201&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Gabriel-SE**
Fix #11319

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -640,11 +640,6 @@ void game::start_game(std::string worldname)
     u.set_highest_cat_level();
     //Calc mutation drench protection stats
     u.drench_mut_calc();
-    if (scen->has_flag("FIRE_START")){
-            m.add_field(u.pos().x + 5, u.pos().y + 3, field_from_ident("fd_fire"), 3 );
-            m.add_field(u.pos().x + 7, u.pos().y + 6, field_from_ident("fd_fire"), 3 );
-            m.add_field(u.pos().x + 3, u.pos().y + 4, field_from_ident("fd_fire"), 3 );
```
Comment (KA101) [inline src/game.cpp]: These offsets are kinda important so the PC doesn't spawn in with a Raging Fire adjacent.  We had that in the initial mergetest version and such starts are unplayable.

(Flashpoint: surrounded by fire on turn 2, On Fire! at turn 3, hoping you saved that char as a template on turn 4.)

**Comment by Gabriel-SE** [comment]
> If I'm reading the replacement code right, any non-wall, non-door, indoor, flammable tile in the building is a valid spot, so the PC could spawn directly onto a raging fire. D8
Wall has "FLAMMABLE" tag, so it's a valid target.
Doors and windows are excluded to give player chance to escape.
Any other terrain or furniture with FLAMMABLE or FLAMMABLE_ASH is valid target.
t_floor doesn't have flammable tag and player usually doesn't spawn on a furniture tile but it's possible to spawn next to a burning wall.

>These offsets are kinda important so the PC doesn't spawn in with a Raging Fire adjacent. We had that in the initial mergetest version and such starts are unplayable.
I can add a safe radius centering from player's location, ex : 5 for exclusion.
Is it acceptable ?

**Comment by KA101** [comment]
Radius 3 would suffice; 5 is fine, too.  Interesting re floors, as fire spreads to 'em just fine.  Door/window exclusion is well thought-out, nice going and sorry I didn't make that clear earlier..

**Comment by Gabriel-SE** [comment]
Added radius 3 as safe area as requested.
t_floor has FLAMMABLE_HARD, so it's not included.

**Pull request merged**